File: comment/src/main/java/musicStreaming/_global/logger/CustomLogger.java
package musicStreaming._global.logger;

import java.util.HashMap;
import java.util.Map;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

// 프로그램에서 일어나는 외부 환경과의 상호작용들을 로깅하기 위해서
public class CustomLogger {
    public static Logger logger = LoggerFactory.getLogger("musicStreaming.comment.custom");
    public static Map<CustomLoggerType, String> customLoggerTypeMap = new HashMap<>() {{
        put(CustomLoggerType.ENTER_EXIT, "ENTER/EXIT");
        put(CustomLoggerType.ENTER, "ENTER");
        put(CustomLoggerType.EXIT, "EXIT");
        put(CustomLoggerType.EFFECT, "EFFECT");
    }};


    public static void debug(CustomLoggerType type, String message, String params) {
        logger.debug(CustomLogger.buildNormalMessages(type, message, params));
    }

    public static void debug(CustomLoggerType type, String message) {
        logger.debug(CustomLogger.buildNormalMessages(type, message, "{}"));
    }

    public static void debug(CustomLoggerType type) {
        logger.debug(CustomLogger.buildNormalMessages(type, "", "{}"));
    }

    public static String buildNormalMessages(CustomLoggerType type, String message, String params) {
        final StackTraceElement[] ste = Thread.currentThread().getStackTrace();
        return String.format("[%s] [%s] %s: %s", ste[3].toString(), CustomLogger.customLoggerTypeMap.get(type), message, params);
    }


    public static void error(Exception e, String message, String params) {
        logger.error(CustomLogger.buildErrorMessages(e, message, params));
    }

    public static void error(Exception e, String message) {
        logger.error(CustomLogger.buildErrorMessages(e, message, "{}"));
    }

    public static void error(Exception e) {
        logger.error(CustomLogger.buildErrorMessages(e, "", "{}"));
    }

    public static String buildErrorMessages(Exception e, String message, String params) {
        final StackTraceElement[] ste = Thread.currentThread().getStackTrace();
        return String.format("[%s] [%s] %s: %s", ste[3].toString(), e.getClass().getName(), e.getMessage() + " / " + message, params) + "\n" + e.getStackTrace().toString();
    }
}

File: mockServers/musicMockServer/app/_global/logger/CustomLogger.py
from flask import current_app
import traceback

from . import CustomLoggerType


customLoggerTypeMap = {
    CustomLoggerType.ENTER_EXIT: "ENTER/EXIT",
    CustomLoggerType.ENTER: "ENTER",
    CustomLoggerType.EXIT: "EXIT",
    CustomLoggerType.EFFECT: "EFFECT"
}


def debug(type:CustomLoggerType, message:str="", params:str="{}") :
    current_app.logger.debug("[{}] [{}] {}: {}".format(traceback.extract_stack()[-2], customLoggerTypeMap[type], message, params))

def error(e:Exception, message:str="", params:str="{}") :
    current_app.logger.error("[{}] [{}] {}: {}".format(traceback.extract_stack()[-2], e.__class__, e.__doc__ + " / " + message, params) + "\n" + ''.join(traceback.format_tb(e.__traceback__)))
